### ISSUE TITLE
Encrypt and hide

### ISSUE THREAD

**Issue by wd4455** [usememos/memos#3503]
### Describe the solution you'd like

I want to use this to write a diary, but I don't want to log in repeatedly. I want to hide some of the diary content when displaying it. Can I add a marker in the text format that requires another password to view, such as:     ||* Hidden that requires a password to display  *||

### Type of feature

User Interface (UI)

**Comment by loockeeer** [comment]
Hi ! I'm looking for the exact same feature ! 
My first question is: should the entire note be "locked" or not ? In my opinion the first option is easier to implement, and a little less messy too, though we could imagine a custom "__encrypted" code-block tag that would trigger a prompt for encryption and decryption ?
I suggest we use the term "locked note(s)" to refer to such objects.
Perhaps a valid implementation would be end-to-end encryption allowing the end-user to decrypt the note when they want to, and adding little to no logic backend-wise (0 backend change if we use the custom markdown trick referred above).

This is just a baseline and I hope you have other suggestions / critics so that we can refine the concept.

**Comment by loockeeer** [comment]
This approach would, though, make migrations hard to follow for the locked notes, depending on what we'd encrypt. (If we encrypt JSON.stringify-ed Node arrays, any modification to that would break the note).
